main/define/src/mill/define/Task.scala

```diff
@@ -819,8 +850,13 @@ class PersistentImpl[+T](
     t: Task[T],
     ctx0: mill.define.Ctx,
     readWriter: RW[_],
-    isPrivate: Option[Boolean]
-) extends TargetImpl[T](t, ctx0, readWriter, isPrivate) {
+    isPrivate: Option[Boolean],
+    generatedSourceRoots: Seq[os.SubPath]
```

Comment:
`deferredGeneratedSourceRoots`? It's a bit annoyingly verbose, but we do want to use the same terminology throughout so it's clear what tasks it is related to